### PR TITLE
Clean Windows approach

### LINKED ISSUE #3: Windows package
As a py37/r36/win64 user it would be great if there were a Windows conda package for `r-arrow`.

I understand Windows support is probably more difficult but I'd like to follow along if there are any issues detailing the problems?

### PR DESCRIPTION
**Pull request by xhochy**
Fixes #3 

Same as #15 but I'm trying to be careful of which hacks are really needed.

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -48,11 +55,11 @@ requirements:
 test:
   commands:
     - $R -e "library('arrow')"           # [not win]
-    - "\"%R%\" -e \"library('arrow')\""  # [win]
+    - "\"%R%\" -e \"library('arrow'); data(mtcars); write_parquet(mtcars, 'test.parquet')\""  # [win]
```
Comment (nealrichardson) [inline recipe/meta.yaml:58]: Why not run the full test suite?

File: recipe/install.libs.R [hunk 2]
```diff
@@ -0,0 +1,5 @@
+src_dir = file.path(R_PACKAGE_SOURCE, "src", fsep = "/")
+dest_dir = file.path(R_PACKAGE_DIR, paste0("libs", R_ARCH), fsep="/")
```
Comment (nealrichardson) [inline recipe/install.libs.R:2]:
```suggestion
src_dir <- file.path(R_PACKAGE_SOURCE, "src", fsep = "/")
dest_dir <- file.path(R_PACKAGE_DIR, paste0("libs", R_ARCH), fsep="/")
```

File: recipe/build_win.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+sed -i -e 's;abs;std::abs;g' $BUILD_PREFIX/Lib/R/library/Rcpp/include/Rcpp/DataFrame.h
+
+# Patch build rules
+sed -i -e 's/@ tmp\.def/@/g' $BUILD_PREFIX/Lib/R/share/make/winshlib.mk
+
+# Rename arrow.dll to lib_arrow.dll to avoid conflicts with the arrow-cpp arrow.dll
+sed -i -e 's/R_init_arrow/R_init_lib_arrow/g' r/src/arrowExports.cpp
+sed -i -e 's/useDynLib(arrow/useDynLib(lib_arrow/g' r/NAMESPACE
+
+# Make exprted symbols visible
```
Comment (nealrichardson) [inline recipe/build_win.sh:23]:
```suggestion
# Make exported symbols visible
```

File: recipe/build_win.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export BUILD_PREFIX_SH=${BUILD_PREFIX//\\//}
+export BUILD_PREFIX_SH=/${BUILD_PREFIX_SH//:}
+
+# Patch Makeconf
+sed -i -e "s;g++;${BUILD_PREFIX_SH}/Library/bin/clang++.exe;g" $BUILD_PREFIX/Lib/R/etc/x64/Makeconf
+sed -i -e "s;nm;${BUILD_PREFIX_SH}/Library/bin/llvm-nm.exe;g" $BUILD_PREFIX/Lib/R/etc/x64/Makeconf
+sed -i -e "s;gnu++11;gnu++14;g" $BUILD_PREFIX/Lib/R/etc/x64/Makeconf
+
+# Patch Rcpp
```
Comment (nealrichardson) [inline recipe/build_win.sh:13]: Is there a reference or references for why these patches are correct, or what they're doing? And now that you have a passing build, are you sure all of them are necessary?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -48,11 +55,11 @@ requirements:
...
 
 about:
   home: https://github.com/apache/arrow
-  license: Apache 2.0
+  license: Apache-2.0
   license_file: LICENSE.txt
   summary: R Integration to 'Apache' 'Arrow'.
```
Comment (nealrichardson) [inline recipe/meta.yaml:64]: FWIW this isn't exactly the Title in `r/DESCRIPTION` (and the version of `DESCRIPTION` commented below is quite stale.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 6]
```diff
@@ -1,36 +0,0 @@
-# This file was generated automatically from conda-smithy. To update this configuration,
```
Comment (nealrichardson) [inline .azure-pipelines/azure-pipelines-linux.yml:1]: Are these yamls supposed to be deleted? (I'm not familiar with the conda-forge build system.)

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/meta.yaml:58]: We normally only run very simple (import) tests in conda-forge as test suits are often very expensive. In this case, I should probably run it as we doing experimental stuff.

File: recipe/build_win.sh (hunk 4 above)
Comment (xhochy) [reply recipe/build_win.sh:13]: The Rcpp patch it is already upstreamed, the other ones are all hacky versions of what could be the correct approach. 

I have made this second PR in addition to #15 as I wanted to make sure I have only the necessary patches. Sadly, I need all of them

File: recipe/meta.yaml (hunk 5 above)
Comment (xhochy) [reply recipe/meta.yaml:64]: The commented version is from the initial recipe creation, we may soon have automated updating from CRAN of these but currently this involves a manual regeneration.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 6 above)
Comment (xhochy) [reply .azure-pipelines/azure-pipelines-linux.yml:1]: Ignore everything in conda-forge except `conda-forge.yml` and things in `recipes`. Everything else is auto-generated.

File: recipe/build_win.sh (hunk 4 above)
Comment (xhochy) [reply recipe/build_win.sh:13]: I still have to find the correct way to override `Makeconf` without touching files in the user's home.

File: recipe/build_win.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+# Patch Rcpp
+# https://github.com/RcppCore/Rcpp/pull/1069
+sed -i -e 's;abs;std::abs;g' $BUILD_PREFIX/Lib/R/library/Rcpp/include/Rcpp/DataFrame.h
```
Comment (nealrichardson) [inline recipe/build_win.sh:7]: Now that that Rcpp PR is merged, can you build a new version for conda and then delete this sed? Would this `sed` be safe once Rcpp with the fix is on conda, or would you end up with `std::std::abs`?

**Pull request merged**